internal/engine/bundle_builder.go

```diff
@@ -33,7 +33,13 @@ type BundleBuilder struct {
 	files   []string
 }
 
+type Bundle struct {
+	Name      string
+	Instances []BundleInstance
+}
+
 type BundleInstance struct {
+	Bundle    *Bundle
```

Comment:
This creates a cyclic dependency between the two type, could we have `Bundle` in  `BundleInstance` as type string?